src/ifcblenderexport/blenderbim/bim/import_ifc.py

```diff
@@ -1309,6 +1310,25 @@ def add_project_to_scene(self):
             self.type_collection.name
         ].hide_viewport = True
 
+    def get_presentation_layers(self):
+        for f in self.file.by_type("IfcPresentationLayerAssignment"):
+            pl = bpy.context.scene.BIMProperties.presentation_layers.add()
+            pl.name = f.Name
+            pl.description = f.Description
+            pl.identifier = f.Identifier
+            if f.is_a() == "IfcPresentationLayerWithStyle":
+                pl.layer_on = f.LayerOn if f.LayerOn is not None else True
+                pl.layer_frozen = f.LayerFrozen if f.LayerFrozen is not None else False
+                pl.layer_blocked = f.LayerBlocked if f.LayerBlocked is not None else False
+
+            for item in f.AssignedItems:
+                guid = item.OfProductRepresentation[0].ShapeOfProduct[0].GlobalId
```

Comment:
`OfProductRepresentation` only works for `IfcRepresentation`, not `IfcRepresentationItem`. This means the import will not work for IFCs exported from the BlenderBIM Add-on, as the export links to the representation items. I just realised this part is actually harder than it seems, see bug #1109. I'll leave the implementation as is for now, but make a note of it that it probably needs to be rewritten.